src/node/interfaces.cpp

```diff
@@ -516,20 +516,27 @@ class ChainImpl : public Chain
     }
     bool haveBlockOnDisk(int height) override
     {
-        LOCK(cs_main);
+        LOCK(::cs_main);
         const CChain& active = Assert(m_node.chainman)->ActiveChain();
         CBlockIndex* block = active[height];
         return block && ((block->nStatus & BLOCK_HAVE_DATA) != 0) && block->nTx > 0;
     }
     CBlockLocator getTipLocator() override
     {
-        LOCK(cs_main);
+        LOCK(::cs_main);
         const CChain& active = Assert(m_node.chainman)->ActiveChain();
         return active.GetLocator();
     }
+    CBlockLocator getActiveChainLocator(const uint256& block_hash) override
+    {
+        LOCK(::cs_main);
+        const CBlockIndex* index = m_node.chainman->m_blockman.LookupBlockIndex(block_hash);
+        if (!index) return {};
+        return Assert(m_node.chainman)->ActiveChain().GetLocator(index);
```

Comment:
In commit "wallet: Save wallet scan progress" (86698868cdb878140c7d6a3909d6aa6f024d8cbf)

Assert on this line isn't really in the ideal place because m_node.chainman is already dereferenced above. Would suggest

```c++
const CBlockIndex* index = chainman().m_blockman.LookupBlockIndex(block_hash);
if (!index) return {};
return chainman().ActiveChain().GetLocator(index);
```

Reply:
Done in https://github.com/bitcoin/bitcoin/pull/25036/commits/32e88f795b5e0663eb15c418d06210eb7828a817. Thanks.